crates/wallet/examples/descriptor_with_plan.rs

```diff
@@ -0,0 +1,479 @@
+        .expect("couldn't create plan");
+
+    // Create an input where we can put the plan data
+    // Add the witness_utxo from the deposit transaction to the input
+    let mut input = psbt::Input {
+        witness_utxo: Some(witness_utxo.clone()),
+        ..Default::default()
+    };
+
+    // Update the input with the generated plan
+    unvault_plan.update_psbt_input(&mut input);
+
+    // Push the input to the PSBT
+    psbt.inputs.push(input);
+
+    // Add a default output to the PSBT
+    psbt.outputs.push(psbt::Output::default());
+
+    // Sign it
+    let secp = Secp256k1::new();
+    let mut sighash_cache = SighashCache::new(&psbt.unsigned_tx);
+    let msg = psbt
+        .sighash_msg(0, &mut sighash_cache, None)
+        .unwrap()
+        .to_secp_msg();
+
+    let unvault_key = Xpriv::from_str(unvault_tprv)
+        .expect("couldn't create unvault key")
+        .private_key;
+    let unvault_sig = secp.sign_ecdsa(&msg, &unvault_key);
+
+    psbt.inputs[0].partial_sigs.insert(
+        bitcoin::PublicKey::from_str(unvault_pk).unwrap(),
+        bitcoin::ecdsa::Signature {
+            signature: unvault_sig,
+            sighash_type: EcdsaSighashType::All,
+        },
+    );
+
+    // Finalize the psbt. Miniscript satisfier checks are run at this point,
+    // and if your transaction doesn't satisfy the descriptor, this will error.
+    psbt.finalize_mut(&secp)
+        .expect("problem finalizing unvault psbt");
+
+    // Once again, we could broadcast the transaction if we wanted to
+    // spend using the unvault path. Spend attempts will fail until
+    // after the absolute block height defined in the timelock.
+    let _my_unvault_tx = psbt.extract_tx().expect("failed to extract unvault tx");
+
+    println!("Congratulations, you've just used a miniscript descriptor with a BDK wallet!");
+    println!("Read the code comments for a walk-through of what happened.")
+}
+
+// Some helper functions to clean up the flow.
+
+// Find the OutPoint by spk, useful for ensuring that we grab the right
+// output transaction to use as input for our spend transaction
+fn get_vout(tx: &Transaction, spk: &Script) -> (OutPoint, TxOut) {
+    for (i, txout) in tx.clone().output.into_iter().enumerate() {
+        if spk == &txout.script_pubkey {
+            return (OutPoint::new(tx.compute_txid(), i as u32), txout);
+        }
+    }
+    panic!("Only call get vout on functions which have the expected outpoint");
+}
+
+fn blank_transaction() -> bitcoin::Transaction {
+    bitcoin::Transaction {
+        version: transaction::Version::TWO,
+        lock_time: absolute::LockTime::ZERO,
+        input: vec![],
+        output: vec![],
+    }
+}
+
+fn blank_transaction_with(lock_time: LockTime) -> bitcoin::Transaction {
+    bitcoin::Transaction {
+        version: transaction::Version::TWO,
+        lock_time,
+        input: vec![],
+        output: vec![],
+    }
+}
+
+fn psbt_with_unsigned_tx(tx: bitcoin::Transaction) -> Psbt {
```

Comment:
There's a method for this [`Psbt::from_unsigned_tx`](https://docs.rs/bitcoin/latest/bitcoin/psbt/struct.Psbt.html#method.from_unsigned_tx) - with a slight difference that instead of pushing onto `psbt.inputs` you would just assign to `psbt.inputs[0]`.

Reply:
I have switched to `Psbt::from_unsigned_tx`, thanks. Just to note,`psbt.inputs.push(foo)` still appears to be there.